src/lib/storage/dictionary_segment.cpp

```diff
@@ -98,26 +98,30 @@ const T DictionarySegment<T>::value_of_value_id(const ValueID value_id) const {
 
 template <typename T>
 ValueID DictionarySegment<T>::lower_bound(const T value) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  auto lower_bound_iterator = std::lower_bound(_dictionary.begin(), _dictionary.end(), value);
```

Comment:
you can use the `dictionary()` function here
```suggestion
  auto lower_bound_iterator = std::lower_bound(dictionary().begin(), dictionary().end(), value);
```